### PR TITLE
S3tc compressed texture support

### PR DESCRIPTION
**Pull request by godyZ**

### CODE REVIEW

**Comment by sergey-shambir** [comment]
Nice change. Looks like you had conflict in xcodeproj and there still are 4 files 'project.pbxproj.{LOCAL, BACKUP, BASE, REMOTE}.23553.pbxproj'.

File: cocos2dx/platform/CCImageCommon_cpp.h [hunk 1]
```diff
@@ -1269,6 +1387,134 @@ bool Image::initWithETCData(const void *data, int dataLen)
...
+
+            width >>= 1;
+            height >>= 1;
+        }
+        _data = new unsigned char [_dataLen];
+    }
+    
+    /* load the mipmaps */
+    
+    int encode_offset = 0;
```
Comment (minggo) [inline cocos2dx/platform/CCImageCommon_cpp.h]: encode_offset -> encodeOffset.

File: cocos2dx/platform/CCImageCommon_cpp.h [hunk 2]
```diff
@@ -1269,6 +1387,134 @@ bool Image::initWithETCData(const void *data, int dataLen)
...
+    int decode_offset = 0;
```
Comment (minggo) [inline cocos2dx/platform/CCImageCommon_cpp.h]: decode_offset -> decodeOffset

File: cocos2dx/platform/CCImageCommon_cpp.h [hunk 3]
```diff
@@ -1269,6 +1387,134 @@ bool Image::initWithETCData(const void *data, int dataLen)
...
+            }
+            
+            _mipmaps[i].address = (unsigned char *)_data + decode_offset;
+            _mipmaps[i].len = (stride * height);
+            memcpy((void *)_mipmaps[i].address, (void *)&decodeImageData[0], _mipmaps[i].len);
+            decode_offset += stride * height;
+
+        }
+        
+		encode_offset += size;
```
Comment (minggo) [inline cocos2dx/platform/CCImageCommon_cpp.h]: wrong indent

**Pull request merged**